sim/midas/src/main/cc/simif.h

```diff
@@ -64,8 +66,41 @@ class simulation_t {
    */
   virtual void simulation_finish() {}
 
+  /**
+   * Return true if the simulation timed out.
+   */
+  virtual bool simulation_timed_out() { return false; }
+
+  /**
+   * Simulation main loop.
+   *
+   * Initialises the simulation, runs it and cleans it up. Prints profiling
+   * info afterwards.
+   *
+   * Returns an exit code from the simulation.
+   */
+  int run();
```

Comment:
So some tests simply do not call `run()` in simulation_run cases where they want fine grained control over when bridges are ticked? Maybe this should be call, "default_run" or "default_simulation_control_loop" or something in that case.